mmdeploy/codebase/mmdet/models/dense_heads/condinst_head.py

```diff
@@ -0,0 +1,359 @@
+        for cls_score in cls_scores
+    ]
+    flatten_bbox_preds = [
+        bbox_pred.permute(0, 2, 3, 1).reshape(batch_size, -1, 4)
+        for bbox_pred in bbox_preds
+    ]
+    flatten_score_factors = [
+        score_factor.permute(0, 2, 3, 1).reshape(batch_size, -1, 1)
+        for score_factor in score_factors
+    ]
+    flatten_param_preds = [
+        param_pred.permute(0, 2, 3, 1).reshape(batch_size, -1, self.num_params)
+        for param_pred in param_preds
+    ]
+    flatten_cls_scores = torch.cat(flatten_cls_scores, dim=1).sigmoid()
+    flatten_bbox_preds = torch.cat(flatten_bbox_preds, dim=1)
+    flatten_score_factors = torch.cat(flatten_score_factors, dim=1).sigmoid()
+    flatten_param_preds = torch.cat(flatten_param_preds, dim=1)
+
+    points = torch.cat(all_level_points)
+    strides = torch.cat(all_level_strides)
+    tl_x = points[..., 0] - flatten_bbox_preds[..., 0]
+    tl_y = points[..., 1] - flatten_bbox_preds[..., 1]
+    br_x = points[..., 0] + flatten_bbox_preds[..., 2]
+    br_y = points[..., 1] + flatten_bbox_preds[..., 3]
+    #
+    bboxes = torch.stack([tl_x, tl_y, br_x, br_y], -1)  # decode
+    scores = flatten_cls_scores
+    score_factors = flatten_score_factors
+    param_preds = flatten_param_preds
+    scores = scores * score_factors
+
+    # get post processing config
+    post_params = get_post_processing_params(deploy_cfg)
+    max_output_boxes_per_class = post_params.max_output_boxes_per_class
+    iou_threshold = cfg.nms.get("iou_threshold", post_params.iou_threshold)
+    score_threshold = cfg.get("score_thr", post_params.score_threshold)
+    pre_top_k = post_params.pre_top_k
+    keep_top_k = cfg.get("max_per_img", post_params.keep_top_k)
+    nms_type = cfg.nms.get("type")
+
+    # Rewrite nms for onnx convertion
+    if with_nms:
+        (
+            batched_dets,
+            batched_labels,
+            batched_param_preds,
+            batched_points,
+            batched_strides,
+        ) = _nms_condinst(
+            self,
+            bboxes,
+            scores,
+            score_factors,
+            param_preds,
+            points,
+            strides,
+            max_output_boxes_per_class,
+            iou_threshold,
+            score_threshold,
+            pre_top_k,
+            keep_top_k,
+            nms_type,
+        )
+
+    results_list = []
+    for dets, labels, param_preds, points, strides in zip(
+        batched_dets,
+        batched_labels,
+        batched_param_preds,
+        batched_points,
+        batched_strides,
+    ):
+        results = InstanceData()
+        results.dets = dets
+        results.bboxes = dets[:, :4]
+        results.scores = dets[:, -1]
+        results.labels = labels
+        results.param_preds = param_preds
+        results.points = points
+        results.strides = strides
+        results_list.append(results)
+    return results_list
+
+
+def _nms_condinst(
```

Comment:
could we refactor and use multiclass_nms , just like rtmde-inst head
https://github.com/open-mmlab/mmdeploy/blob/01a88bee407029a2f0fea8fcb68597e6ae9bfd77/mmdeploy/codebase/mmdet/models/dense_heads/rtmdet_ins_head.py#L105